asm/c128/stubs_test.go

```diff
@@ -0,0 +1,231 @@
+		a:   1 + 2i,
+		dst: []complex128{0, 0, 0},
+		x:   []complex128{0, 0},
+		y:   []complex128{1, 1, 1},
+		ex:  []complex128{1, 1}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 + 2i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i},
+		y:   []complex128{1, 2, 1},
+		ex:  []complex128{-1 + 1i, 1i, -1 + 1i}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -1i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i},
+		y:   []complex128{1, 2, 1},
+		ex:  []complex128{2, 3, 2}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -1i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i, 1i, 1i}[1:4],
+		y:   []complex128{1, 1, 2, 1, 1}[1:4],
+		ex:  []complex128{2, 3, 2}},
+	{incX: 2, incY: 4, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -2,
+		dst: []complex128{1i, 1i, 1i, 1i, 1i},
+		x:   []complex128{2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i},
+		y:   []complex128{1, 1, 2, 1, 1},
+		ex:  []complex128{-3 - 2i, -3 - 2i, -2 - 2i, -3 - 2i, -3 - 2i}},
+	// Run big test twice, once aligned once unaligned.
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: -2, incY: -2, incDst: -3, ix: 18, iy: 18, idst: 27,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: -2, incY: 2, incDst: -3, ix: 18, iy: 0, idst: 27,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+}
+
+func guardVector(v []complex128, g complex128, g_ln int) (guarded []complex128) {
+	guarded = make([]complex128, len(v)+g_ln*2)
+	copy(guarded[g_ln:], v)
+	for i := 0; i < g_ln; i++ {
+		guarded[i] = g
+		guarded[len(guarded)-1-i] = g
+	}
+	return guarded
+}
+
+func isValidGuard(v []complex128, g complex128, g_ln int) bool {
+	for i := 0; i < g_ln; i++ {
+		if v[i] != g || v[len(v)-1-i] != g {
+			return false
+		}
+	}
+	return true
+}
+
+func TestAxpyUnitary(t *testing.T) {
+	var x_gd, y_gd complex128 = 1, 1
+	for j, v := range tests {
+		xg_ln, yg_ln := 4+j%2, 4+j%3
```

Comment:
The exact value of the guard length isn't all that important, odd vs even (plus mismatch) is all the information that it brings to the test.  I was using two `bool` elements in the test struct, but I switched to this mod setup in order to clean up the test code (two if blocks replaced by this single line assignment).  If we had a C-style ternary, it would be compressed into `xg_ln := 4+(v.x_aligned?0:1)`.    I chose 4 because it's the width of a `ymm` register, so it wouldn't change when adding AVX code.

In the end, I chose to keep the test struct as simple as possible.  Of course it's not set in stone, but I don't think we need to add it to the struct/table unless we can ensure that it isn't the same across all tests.